sstables/sstables.cc

```diff
@@ -1396,6 +1396,26 @@ void sstable::write_filter() {
     write_simple<component_type::Filter>(filter_ref);
 }
 
+uint64_t sstable::total_components_memory_size() const {
+    return _components->filter ? _components->filter->memory_size() : 0;
+}
```

Comment:
should we reserve 10% for filters or all in-memory metadata?  IIRC, we reserve 5% for summary (2000:1 ratio), we also have compression metadata, so possibly it's ok to start by limiting  filters to 10% of memory. otherwise, in practice, we might be essentially disabling filters.

a better name may be total_reclaimable_memory_size()

Reply:
I have updated the method name and the PR based on the reclaimable concept. We now track reclaimable memory and we reclaim when it crosses the reclaim threshold. This makes it more clear than the previous terminologies I used. Thank you!